src/platform_specific/named_threads.unix.cc

```diff
@@ -0,0 +1,45 @@
+#include "named_threads.h"
+
+#include <cassert>
+#include <type_traits>
+#include <vector>
+
+#include <pthread.h>
+
+namespace celerity::detail {
+
+static_assert(std::is_same_v<std::thread::native_handle_type, pthread_t>, "Unexpected native thread handle type");
+
+constexpr auto PTHREAD_MAX_THREAD_NAME_LEN = 16;
+
+static inline void set_thread_name_unix(const pthread_t thread_handle, const std::string& name) {
+	auto truncated_name = name;
+	truncated_name.resize(PTHREAD_MAX_THREAD_NAME_LEN - 1); // -1 because of null terminator
+	[[maybe_unused]] const auto res = pthread_setname_np(thread_handle, truncated_name.c_str());
+	assert(res == 0 && "Failed to set thread name");
+}
+
+static inline std::string get_thread_name_unix(const pthread_t thread_handle) {
+	auto name = std::vector<char>(PTHREAD_MAX_THREAD_NAME_LEN, '\0');
+	[[maybe_unused]] const auto res = pthread_getname_np(thread_handle, name.data(), name.size());
+	assert(res == 0 && "Failed to get thread name");
+	return name.data(); // Strip null terminator
```

Comment:
What happens here would be more obvious when using
```c++
char name[PTHREAD_MAX_THREAD_NAME_LEN] = {};
```
instead of the char vector (also needs one fewer include)

Reply:
I don't necessarily agree about the regular array being more obvious, but I have no strong opinion about this either way, so I changed it to a normal array.